osm_fudge/index/bk_tree.py

```diff
@@ -4,48 +4,87 @@
 '''
 
 
+def grouped(iterable, n):
+    return zip(*[iter(iterable)]*n)
+
+
 class BKTree:
     '''A bk-tree implementation and defined user methods
        for insertion and querying the string
     '''
 
     def __init__(self, distanceMetrics):
-        self.tree = {}
+        self.tree = [[]]
```

Comment:
Hi @daniel-j-h 👋 , have made the changes to do it list way. It's done in insertion method, but lookup method needs to refactored for the list implementation. Need an 👁 here. It's a little messy atm, will clean it up sooner.